### ISSUE TITLE
How to run YCSB against remote HBase

### ISSUE THREAD

**Issue by gioargyr** [brianfrankcooper/YCSB#886]
I have a 1+2 cluster running Hadoop-2.5.2 and HBase-1.2.3 (both fully distributed) running with no problems at all(verified). I downloaded and built the most recent version of YCSB (0.13.0) to run it against HBase! I also made the "usertable" table with "family" column.

    [1st case]I run YCSB from the Hadoop-HBase-master-VM with this command: ./bin/ycsb load hbase10 -P workloads/workloada -p columnfamily=family and everything works fine.
    [2nd case] I want to have another VM which runs exclusively YCSB (a ycsb-client-VM). But, when I try to run YCSB from another VM I can't manage to run the afforementioned command succesfully. I tried with a lot of ways (I have already created /conf dir inside hbase10-binding and stored hbase-site.xml in it), and each time a get a different error which will make the post huge if I mention all of them here.

Has anyone run YCSB against HBase from a different server than HBase is running?

Is there any tutorial on how to run YCSB from dedicated ycsb-client-vm?

**Comment by risdenk** [comment]
You need to make sure you have the hbase/conf directory on the YCSB classpath. That would mean copying the conf directory used for HBase to the YCSB client machine. It sounds like you tried to do something like this.

Can you share the specific commands tried and errors received?

**Comment by gioargyr** [comment]
There is an hbase10-binding directory, but there wasn't any hbase10-binding/conf directory. So, according to this: https://github.com/brianfrankcooper/YCSB/tree/master/hbase10 (7th step), I created the hbase10-binding/conf directory. After that, I stored only hbase-site.xml in it. As I understand from what you say, I need to store all the files from hbase/conf (hbase-master-VM) in hbase10-binding/conf (in ycsb-client-VM), . Please, verify if I understand what you say.

The last error I keep getting is "`java.lang.ArrayIndexOutOfBoundsException`", when I run the command: `./bin/ycsb load hbase10 -P workloads/workloada -p columnfamily=family`

**Comment by risdenk** [comment]
Share the full stack traces. That would give a full picture of what is happening.

On your HBase machine, there is a conf directory with hbase-site.xml and others. Copy that folder to a location on your YCSB machine. It doesn't have to be under the binding directory. When you run  YCSB do the following:

`./bin/ycsb load hbase10 -cp PATH_TO_HBASE_CONF_DIR -P workloads/workloada -p columnfamily=family`

This is outlined here: https://github.com/brianfrankcooper/YCSB/tree/master/hbase098#4-run-the-workload

**Comment by gioargyr** [comment]
Hello again!
I just managed (I was sooo late because we had several issues with my VMs powered by OpenStack in my lab the last days!) to try what you suggested and everything went fine!
Of course I got the well-known warning "java.io.IOException: No FileSystem for scheme: hdfs" but as I see in other issues it isn't important and I can get rid of it if I want.
Just to mention that I had to add the alias-names of the hbase-cluster-vms with their ips to the /etc/hosts of the dedicated ycsb-client-vm.
Thank you very much. I was really confused with the hbase-conf thing and there is literally no tutorial or anything in the whole internet regarding this issue.
(Happy New Year!)
